src/tool_operate.c

```diff
@@ -687,22 +687,222 @@ static CURLcode post_per_transfer(struct GlobalConfig *global,
   return result;
 }
 
+static char *ipfs_gateway(void)
+{
+  char *gateway;
+  char *ipfs_path;
+  char *gateway_composed_file_path;
+  FILE *gateway_file;
+
+  gateway = curlx_getenv("IPFS_GATEWAY");
+
+  /* Gateway is found from environment variable. */
+  if(gateway) {
+    bool add_slash = (gateway[strlen(gateway) - 1] == '/') ? FALSE : TRUE;
```

Comment:
Fixed (by changing it to `if(gateway && strlen(gateway)) {`)

Reply:
That makes unnecessarily 2 `strlen` calls...

Reply:
@jzakrzewski I see. Any hint on what a proper fix is there? I can't do the strlen earlier as that will crash if `gateway` = `null` (tried it). Sure, i can put it in a scoped variable, but then i'm wondering what's better. one extra strlen or another if checking for length?

Reply:
I'm not sure what curl style guide says, but `if(gateway && *gateway)` is one option. Otherwise there's also "the long option":
```c
if(gateway) {
  size_t len = strlen(gateway);
  bool add_slash = ((len > 0) && (gateway[len - 1] == '/')) ? FALSE : TRUE;
```
or some other similar combination